src/components/modal/createModal/CreateModal.js

```diff
@@ -4,68 +4,169 @@ import { Modal, Checkbox, BasicInput, Dropdown } from '../../index';
 import { strings } from '../../../utils';
 import { identifiers } from '../../../constants';
 import './createModal.css';
+import { regexes } from '../../../constants';
 
 const plus = require('../../../assets/images/plus.svg');
 
 class CreateModal extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      createParametersState: this.initializeCreateParameters(),
-      showError: false,
+      inputs: this.initializeCreateParameters(),
+      showErrorMessage: false,
       createPassword: true,
     };
   }
 
   initializeCreateParameters = () => {
-    let createParametersState = {};
+    let inputs = {};
     this.props.createParameters.forEach(item => {
-      if (item.defaultValue) {
-        createParametersState[item.id] = item.defaultValue;
-      } else if (item.options && item.options.length > 0) {
-        createParametersState[item.id] = item.options[0].key;
-      }
-      else {
-        createParametersState[item.id] = item.type === 'boolean' ? false : '';
+      if (item.type === 'email') {
+        inputs[item.id] = { value: '', validation: 'email', isValid: true, errorMessage: '' };
+      } else if (item.type === 'text') {
+        inputs[item.id] = { value: '', validation: 'text', isValid: true, errorMessage: '' };
+      } else if (item.type === 'password') {
+        inputs[item.id] = { value: '', validation: this.createPassword ? 'password' : 'none', isValid: true, errorMessage: '' };
+      } else if (item.type === 'select' && item.options.length > 0) {
+        inputs[item.id] = { value: item.options[0].key, validation: 'none' };
+      } else if (item.type === 'boolean') {
+        inputs[item.id] = { value: item.value, validation: 'none' };
       }
     });
-    return createParametersState;
+    return inputs;
   }
 
   setCreateParameters = () => {
-    this.setState({ createParametersState: this.initializeCreateParameters(), showError: false });
+    this.setState({ inputs: this.initializeCreateParameters(), showErrorMessage: false });
   }
 
   create = async () => {
-    this.setState({ showError: true });
-    if (this.state.createPassword) {
-      return this.props.create({ ...this.state.createParametersState, password: undefined }, this.state.createPassword);
+    let toValidate = 0;
+    let validated = 0;
+    let inputs = this.state.inputs;
+    for (let key in inputs) {
+      if (inputs.hasOwnProperty(key) && inputs[key].validation !== 'none') {
+          toValidate++;
+          let validation = this.validate(inputs[key].validation, inputs[key].value);
+          inputs[key].isValid = validation.isValid;
+          inputs[key].errorMessage = validation.errorMessage;
+          if(validation.isValid) {
+            validated++;
+          }
+      } 
+    }
+
+    if (toValidate === validated) {
+      for (let key in inputs) {
+        if (inputs.hasOwnProperty(key)) {
+          inputs[key] = inputs[key].value;
```

Comment:
reassign to a different variable instead of mutating the original

Reply:
looping over objects in object here